pilot/docker/Dockerfile.sidecar_injector

```diff
@@ -8,7 +8,7 @@ COPY sidecar-injector /usr/local/bin/
 # 1001 is chosen as the UID to be added to the istio group as this is the UID that sidecar-injector in the docker container runs as.
 RUN addgroup --system --gid 1001 istio && \
```

Comment:
I think I left the comment in other PR - please don't add new groups in the docker file. Either add it to the base, or even better use the 1331 UID we already use for the istio user. 

It is quite confusing since in 'deb' file we add the istio user with the UID of the proxy. 

Is there any particular reason to use 1001 and create even more inconsistency ?

Reply:
@clyang82 can you take on addressing the adduser points @costinm has raised in a separate PR?  I think the simple answer is to use 1331, however, this has potential to break distroless.  I think adding an inconsistent UID/GID doesn't make sense.  Please use one UID/GID throughout dockerfiles.  Currently our operators expect UID/GId to be 1331.

 If the go code itself is using "istio" in some way, that will need to be addressed.

Try not to break distroless as you do the work :)

Cheers
-steve

Reply:
I just searched in operator and istio repo, did not find the `UID:1331`. Could you point me out? yes. we should keep consistent if we already have 1331. I just checked gcr.io/distroless/static:nonroot the nonroot UID is 65532

Reply:
I was mistaken on the UID.  The proper UID = 1337.  See: https://github.com/istio/istio/blob/master/docker/Dockerfile.base#L31-L33